File: baekjoon/Implementation/10866.cpp
// 백준 10866번 : 덱
#include <iostream>
#include <deque>

using namespace std;

int main() {
    deque<int> dq;

    int N;
    cin >> N;

    for (int i = 0; i < N; i++) {
        string command;
        cin >> command;

        int num;
        if (command == "push_back") {
            cin >> num;
            dq.push_back(num);
        } else if (command == "push_front") {
            cin >> num;
            dq.push_front(num);
        } else if (command == "pop_front" || command == "front") {
            if (!dq.empty()) {
                cout << dq.front() << '\n';
                if (command == "pop_front") {
                    dq.pop_front();
                }
            } else {
                cout << -1 << '\n';
            }
        } else if (command == "pop_back" || command == "back") {
            if (!dq.empty()) {
                cout << dq.back() << '\n';
                if (command == "pop_back") {
                    dq.pop_back();
                }
            } else {
                cout << -1 << '\n';
            }
        } else if (command == "size") {
            cout << dq.size() << '\n';
        } else if (command == "empty") {
            if (!dq.empty()) {
                cout << 0 << '\n';
            } else {
                cout << 1 << '\n';
            }
        }
    }
    return 0;
}

File: baekjoon/DataStructure/10866.py
# 백준 10866번 : 덱
import sys
input = sys.stdin.readline

if __name__ == "__main__":
    N = int(input())

    st = []
    for _ in range(N):
        cmd = input().split()

        if cmd[0] == "push_front":
            st.insert(0, cmd[1])
        elif cmd[0] == "push_back":
            st.append(cmd[1])
        elif cmd[0] == "pop_front":
            if st:
                print(st[0])
                st.pop(0)
            else:
                print(-1)      
        elif cmd[0] == "pop_back":
            if st:
                print(st[-1])
                st.pop()
            else:
                print(-1)
        elif cmd[0] == "size":
            print(len(st))
        elif cmd[0] == "empty":
            if st:
                print(0)
            else:
                print(1)
        elif cmd[0] == "front":
            if st:
                print(st[0])
            else:
                print(-1)
        elif cmd[0] == "back":
            if st:
                print(st[-1])
            else:
                print(-1)